src/ziggurat/messaging/producer.clj

```diff
@@ -72,10 +72,87 @@
        (sentry/report-error sentry-reporter e
                             "Pushing message to rabbitmq failed, data: " message-payload)))))
 
+(defn- get-channel-retry-count [topic-entity channel]
+  (-> (ziggurat-config) :stream-router topic-entity :channels channel :retry :count))
+
+(defn- get-channel-retry-queue-timeout-ms [topic-entity channel]
+  (-> (ziggurat-config) :stream-router topic-entity :channels channel :retry :queue-timeout-ms))
+
+(defn- channel-retries-exponential-backoff-enabled [topic-entity channel]
+  "Get exponential backoff enabled for specific channel from config."
+  (-> (ziggurat-config) :stream-router topic-entity :channels channel :retry :exponential-backoff-enabled))
```

Comment:
it will be better if we refactor the config - instead of `:exponential-backoff-enabled` and `:exponential-backoff-enabled` we should have `{:exponential-backoff {:enabled true :count 4}}`

Reply:
this is updated, changed the config structure for exponential backoff